### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by Paskab012**
### Hello😊
In this PR I have made implemented the use of redux inside react. Follow  the steps below to see the Caneva I used to implement changes.

## Changes Made
- First of all -  keep the state immutable in reducers!
- Secondly, changed the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
- Use useSelector to consume the state.
- In your React component responsible for adding new books, add funcionality for "Add new" button click:
    - Import Redux useDispatch.
    - Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
    - Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for removing books, add a funcionality for "Remove" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for categories, add a funcionality for "Check status" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 import Book from './Book';
 import AddBook from './Addbook';
 
-function Books({ books }) {
+function Books() {
+  const books = useSelector((state) => state.booksReducer);
+
   return (
     <>
       {books.map((book) => (
```
Comment (medaminedev66) [inline src/components/Books.js]: - [ ] When I click on `add book`, an error shows up on the console. Please make sure that the books list is fetched from the Redux store and rendered on the Books layout.

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @KABIKA681  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors ✔️
- You have a descriptive PR ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const initialState = [{
 export default function booksReducer(state = initialState, action = {}) {
   switch (action.type) {
     case ADD_BOOK:
-      return { ...state, book: action.payload };
+      return [...state, action.payload];
     case REMOVE_BOOK:
       return [...state.filter((book) => book.id !== action.payload.id)];
```
Comment (Kingstalux) [inline src/redux/books/books.js:20]: - [ ] (OPTIONAL) Great job so far 👍 I can now add a book but I cannot remove a book. Kindly fix this issue so as to have the correct functionality of the app 💯

**Review by Kingstalux — APPROVED** [review]
Hi @KABIKA681,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**